src/iptc.cpp

```diff
@@ -110,6 +195,15 @@ namespace Exiv2 {
 
     std::ostream& Iptcdatum::write(std::ostream& os, const ExifData*) const
     {
+        const StringValue* val = dynamic_cast<const StringValue*>(&value());
+        if (val != NULL) {
+            // In case of being a StringValue, it could have non UTF-8 which we want to supress
+            if (is_utf8(val->toString())) {
+                return os << value();
+            } else {
+                return os << N_("(Non UTF-8 bytes supressed)");
+            }
+        }
         return os << value();
```

Comment:
Won't this output the value independently whether the string is utf-8 or not?

Reply:
When it's a string, it will fall inside the if block (lines 199-206) and return there. We only reach line 207 in case the value() is not an string. But I think I might now need to do the `dynamic_cast` if when then value is not a StringValue, the method toString() returns an empty string. I will work a bit more there.

Reply:
Oops, I completely missed the return there.. Sorry.